File: 347_top_k_frequent_elements.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer[]}
def top_k_frequent(nums, k)
  counts = Hash.new(0)
  
  nums.each { |num| counts[num] += 1 }
  
  counts.to_a.sort { |pair1, pair2| pair2.last <=> pair1.last }.map{ |pair| pair.first }[0...k]
end

File: 347_top_k_frequent_elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
  let counts = {};
  
  for (let i = 0; i < nums.length; ++i) {
    if (counts[nums[i]]) {
      counts[nums[i]]++;
    } else {
      counts[nums[i]] = 1;
    }
  }
  
  let keys = Object.keys(counts);
  keys.sort((a, b) => {
    return counts[b] - counts[a];
  });
  
  return keys.map((el) => parseInt(el) ).slice(0, k);
};